libuserhelper.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) 2020 Jolla Ltd.
+ * Copyright (c) 2020 Open Mobile Platform LLC.
+ *
+ * All rights reserved.
+ *
+ * BSD 3-Clause License, see LICENSE.
+ */
+
+#include "libuserhelper.h"
+
+#include <libuser/user.h>
+
+#include <QDebug>
+
+LibUserHelper::LibUserHelper()
+{
+}
+
+uint LibUserHelper::addGroup(const QString &group)
+{
+    struct lu_error *error = nullptr;
+    struct lu_context *context = lu_start(NULL, lu_user, NULL, NULL, lu_prompt_console, NULL, &error);
+
+    if (!context) {
+        qWarning() << "Error creating context:" << lu_strerror(error);
+        lu_error_free(&error);
+        return 0;
```

Comment:
Should this be `LU_VALUE_INVALID_ID` since it's obviously a failure? Same for `rv` in case if `lu_group_default()` fails.

Reply:
I'd rather not add libuser dependencies to SailfishUserManager class now that they're all moved to separate helper class. That zero should be safe enough error value as it's reserved for root anyway.

Reply:
If zero represents an error, then `LU_VALUE_INVALID_ID` returned by `lu_ent_get_first_id()` needs to be replaced with zero, too. And if it returns zero.... then I don't know :) How about returning a positive signed int on success and `-1` on error?

Reply:
Yep, missed that possible error situation. Now returns zero in that case.